src/index.js

```diff
@@ -0,0 +1,159 @@
+    this.known = [];
+  }
+
+  getColor() {
+    const { distance, knownMax } = this.options;
+    const h = this._pickHue();
+    const s = this._pickSaturation(h);
+    const b = this._pickBrightness(h, s);
+    const hsl = this._HSVtoHSL(h, s, b);
+    const PASSABLE_DISTANCE = typeof distance !== 'undefined' ? distance : 37; 
+    const KNOWN_MAX = typeof knownMax !== 'undefined' ? knownMax : 20; 
+
+    // Detect color similarity. If values are too close to one another, call
+    // getColor until enough dissimilarity is acheived.
+    if (this.known.length &&
+      this.known.length <= KNOWN_MAX &&
+      this.known.some(v => this._euclideanDistance(v, hsl) < PASSABLE_DISTANCE)) {
+      return this.getColor();
+    } else {
+      this.known.push([...hsl]);
+      return this._colorWithModifiers(h, s, b) 
+    }
+  }
+
+  _colorWithModifiers = (h, s, b) => {
+    const clamp = (n, min, max) => n <= min ? min : n >= max ? max : n;
+    const { brightness, saturation } = this.options;
+
+    // Modify brightness/saturation if provided
+    s = saturation ? clamp(s + saturation, 0, 100) : s;
+    b = brightness ? clamp(b + brightness, 0, 100) : b;
+
+    // re-run conversion accounting for post modifications to s and b.
+    const hsl = this._HSVtoHSL(h, s, b);
+
+    return {
+      hsl: {
+        raw: hsl,
+        formatted: `hsl(${hsl[0]}, ${hsl[1]}%, ${hsl[2]}%)`
+      }
+    };
+  }
+
+  _pickHue = () => {
+    let hue = this._pseudoRandom([0, this.HUE_MAX]);
+    const min = hue % this.HUE_MAX;
+    const max = (hue + 1) % this.HUE_MAX;
+
+    hue = this._pseudoRandom([min, max]);
+
+    // Red is on both ends of the color spectrum. Instead of storing red as two
+    // ranges, lookup is grouped in colorDictionary using negative numbers.
+    if (hue < 0) hue = this.HUE_MAX + hue;
+
+    return hue;
+  }
+
+  _pickSaturation = (h) => {
+    const saturationRange = this._getColorInfo(h)[2];
+    let min = saturationRange[0];
+    let max = saturationRange[1];
+    return this._pseudoRandom([min, max]);
+  }
+  
+
+  _pickBrightness = (h, s) => {
+    let min = this._getMinimumBrightness(h, s);
+    let max = 100;
+    return this._pseudoRandom([min, max]);
+  }
+
+  _getMinimumBrightness = (h, s) => {
+    const lowerBounds = this._getColorInfo(h)[1];
+    for (let i = 0; i < lowerBounds.length - 1; i++) {
+      const s1 = lowerBounds[i][0];
+      const v1 = lowerBounds[i][1];
+      const s2 = lowerBounds[i + 1][0];
+      const v2 = lowerBounds[i + 1][1];
+      if (s >= s1 && s <= s2) {
+        const m = (v2 - v1) / (s2 - s1);
+        const b = v1 - m * s1;
+        return m * s + b;
+      }
+    }
+    return 0;
+  }
+
+  // A linear congruential generator (LCG) algorithm that yields a sequence of
+  // pseudo-randomized numbers calculated with a discontinuous piecewise linear
+  // equation. see: indiegamr.com/generate-repeatable-random-numbers-in-js
+  _pseudoRandom = (range) => {
```

Comment:
I would call this something like _numberGenerator rather than pseudoRandom since it's not random.

Reply:
Isn't it a pseudorandom generator? https://en.wikipedia.org/wiki/Pseudorandom_number_generator